docs/declarative-setup.md

```diff
@@ -0,0 +1,130 @@
+
+Example:
+
+```yaml
+apiVersion: v1
+data:
+  dex.config: |
+    connectors:
+    - type: github
+      id: github
+      name: GitHub
+      config:
+        clientID: e8f597564a82e99ba9aa
+        clientSecret: e551007c6c6dbc666bdade281ff095caec150159
+  repositories: |
+    - passwordSecret:
+        key: password
+        name: my-secret
+      url: https://github.com/argoproj/my-private-repository
+      usernameSecret:
+        key: username
+        name: my-secret
+  url: http://localhost:4000
+kind: ConfigMap
+metadata:
+  name: argocd-cm
+```
+
+## Clusters
+
+Cluster credentials are stored in secrets same as repository credentials but does not require entry in `argocd-cm` config map. Each secret must have label
+`argocd.argoproj.io/secret-type: cluster` and name which is following convention: `<hostname>-<port>`.
+
+The secret data must include following fields:
+* `name` - cluster name
+* `server` - cluster api server url
+* `config` - JSON representation of following data structure:
+
+```yaml
+# Basic authentication settings
+username: string
+password: string
+# Bearer authentication settings
+bearerToken: string
+# IAM authentication configuration
+awsAuthConfig:
+    clusterName: string
+    roleARN: string
+# Transport layer security configuration settings
+tlsClientConfig:
+    # PEM-encoded bytes (typically read from a client certificate file).
+    caData: string
+    # PEM-encoded bytes (typically read from a client certificate file).
+    certData: string
+    # Server should be accessed without verifying the TLS certificate
+    insecure: boolean
+    # PEM-encoded bytes (typically read from a client certificate key file).
+    keyData: string
+    # ServerName is passed to the server for SNI and is used in the client to check server
+    # ceritificates against. If ServerName is empty, the hostname used to contact the
+    # server is used.
+    serverName: string
+
+```
+
+
+Cluster secret example:
+
+```yaml
+apiVersion: v1
+stringData:
+  config: |||
+    {
+        "bearerToken": "<authentication token>",
+        "tlsClientConfig": {
+            "insecure": false,
+            "caData": "<base64 encoded certificate>"
+        }
+    }
+  |||
+  name: mycluster.com
+  server: https://mycluster.com
+kind: Secret
+metadata:
+  labels:
+    argocd.argoproj.io/secret-type: cluster
+  name: mycluster.com-443
+type: Opaque
+```
+
+## SSO & RBAC
+
+* SSO configuration details: [SSO](sso.md)
+* RBAC configuration details: [RBAC](rbac.md)
+
+## Manage Argo CD using Argo CD
+
+Argo CD is able to manage itself since all settings are represented by Kubernetes manifests. The suggested way is to create [Kustomize](https://github.com/kubernetes-sigs/kustomize)
+based application which uses base Argo CD manifests from https://github.com/argoproj/argo-cd and apply required changes on top.
+
+Example of `kustomization.yaml`:
+
+```yaml
+bases:
+- github.com/argoproj/argo-cd//manifests/cluster-install?ref=v0.10.0
```

Comment:
You have an extra / in url: `argo-cd//manifests`

Reply:
Also, should we change the ref to the latest version: v0.10.6?

Reply:
changed to v0.10.6 . After v0.11.0 release lets introduce moving tag `stable` and use it in all docs

Reply:
The extra / is intentional. Kustomize uses it to indicate end of repo url and start of relative path inside